app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CommentsController < ApplicationController
+  before_action :set_repository
+  respond_to :js, :html
+
+  # POST /repositories/1/comments
+  # POST /repositories/1/comments.json
+  def create
+    @comment = @repository.comments.new(comment_params)
+    @comment.author = current_user
+    authorize @comment
+
+    if @comment.save
+      @repository.create_activity :create_comment,
+                                  owner:      current_user,
+                                  parameters: {
```

Comment:
not if the activity.trackable is no longer there ;) That is, even if a repo gets removed (by, for example, the catalog job), the activity is still reachable.